### PR TITLE
JAVA-2276: Document how to configure mapper with Lombok

### PR DESCRIPTION
**Pull request by olim7t**

### CODE REVIEW

File: manual/mapper/config/README.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ configuration (make sure you use version 3.5 or higher):
       <artifactId>maven-compiler-plugin</artifactId>
       <version>3.8.1</version>
       <configuration>
+        <source>1.8</source> <!-- (or higher) -->
+        <target>1.8</target> <!-- (or higher) -->
```
Comment (olim7t) [inline manual/mapper/config/README.md:37]: Not directly related, but I noticed the build fails without this, better be copy-paste safe.

File: manual/mapper/config/README.md [hunk 2]
```diff
@@ -76,3 +78,95 @@ dependencies {
...
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@EqualsAndHashCode
+@ToString
+public class Product {
+  @PartitionKey @Getter @Setter private int id;
```
Comment (tomekl007) [inline manual/mapper/config/README.md:103]: If we are saying that ordering of annotation processors then maybe in this example we should follow the same ordering? so firstly Lombok annotations, and them ours:
```
@EqualsAndHashCode
@ToString
@Entity
public class Product {
 @Getter @Setter @PartitionKey private int id;
```

File: manual/mapper/config/README.md [hunk 3]
```diff
@@ -76,3 +78,95 @@ dependencies {
...
+
+The mapper can process Lombok-annotated classes just like regular code. The only requirement is that
+Lombok's annotation processor must run *before* the mapper's.
+
+With Maven, declaring Lombok as a provided dependency is not enough; you must also redeclare it in
+the `<annotationProcessorPaths>` section, before the mapper:
+
+```xml
+<properties>
+  <java-driver.version>4.0.1-SNAPSHOT</java-driver.version>
```
Comment (tomekl007) [inline manual/mapper/config/README.md]: should we use SNAPSHOT in docs?

File: manual/mapper/config/README.md (hunk 2 above)
Comment (olim7t) [reply manual/mapper/config/README.md:103]: The declaration order of the annotations does not matter, so I see no reason to imply that in our examples.

File: manual/mapper/config/README.md (hunk 3 above)
Comment (olim7t) [reply manual/mapper/config/README.md]: Good catch, I will fix that.

**Pull request merged**